migrations/2024-12-17-144959_community-post-tags/up.sql

```diff
@@ -0,0 +1,26 @@
+-- a tag for a post, valid in a community. created by mods of a community
+CREATE TABLE tag (
+    id serial PRIMARY KEY,
+    ap_id text NOT NULL UNIQUE,
+    name text NOT NULL,
+    published timestamptz NOT NULL DEFAULT now(),
+    updated timestamptz,
+    deleted timestamptz
+);
+
+-- indicates this tag was created by the mod of a community and can be applied to posts in this community
+CREATE TABLE community_post_tag (
+    community_id int NOT NULL REFERENCES community (id) ON UPDATE CASCADE ON DELETE CASCADE,
+    tag_id int NOT NULL REFERENCES tag (id) ON UPDATE CASCADE ON DELETE CASCADE,
+    published timestamptz NOT NULL DEFAULT now(),
+    PRIMARY KEY (community_id, tag_id)
+);
+
+-- an association between a post and a tag. created/updated by the post author or mods of a community
+CREATE TABLE post_tag (
```

Comment:
I can't think of a good naming scheme that could potentially include scope. IE how do we differentiate post tagged by a community, vs a post potentially tagged by the instance? Or a community tagged by the instance, vs the community's tags it creates for its own posts? 

Maybe `post_tag_by_community`

Reply:
i think the better approach here would be to keep this association table "generic" for all post-tag associations, and add additional fields to it to mention the type of association if we need it later. 

the comment just refers to what it's for now, but that purpose can become more generic in the future